lib/fake_artist/table.ex

```diff
@@ -33,85 +53,209 @@ defmodule FakeArtist.Table do
-    %{seats: seats, active_seat: active_seat} = FakeArtist.Game.choose_category(state_pid)
-    players = assemble_players(seats, id_map, active_seat)
+  def handle_call(
+        :progress_game,
+        _from,
+        state = %{
+          active_players: active_players,
+          players: players,
+          remaining_turns: remaining_turns,
+          table_name: table_name
+        }
+      ) do
+    if remaining_turns == 1 do
+      Logger.info(fn -> "Voting." end)
+
+      state = %{
+        state
+        | little_state: :vote
+      }
+
+      {:reply, :ok, state}
+    else
+      state = %{
+        state
+        | active_players: get_active_players(players, active_players),
+          remaining_turns: remaining_turns - 1
+      }
 
-    FakeArtistWeb.Endpoint.broadcast("table:#{table_name}", "update_game", %{players: players})
+      FakeArtistWeb.Endpoint.broadcast("table:#{table_name}", "update", state)
 
-    {:reply, :ok, {id_map, table_name, user_count, state_pid}}
+      {:reply, :ok, state}
+    end
   end
 
   def handle_info(
         {:DOWN, _ref, :process, _from, _reason},
-        {id_map, table_name, user_count, _old_state_pid}
+        state = %{connected_computers: connected_computers}
       ) do
     Logger.info(fn -> "table lost connection." end)
-    Logger.info(fn -> "player count decreased from #{user_count} to #{user_count - 1}" end)
 
-    user_count = user_count - 1
+    Logger.info(fn ->
+      "player count decreased from #{connected_computers} to #{connected_computers - 1}"
+    end)
 
-    if user_count == 0 do
+    if connected_computers - 1 == 0 do
       Logger.info(fn -> "suicide" end)
-      {:stop, :shutdown, {%{}, "", 0}}
+      {:stop, :shutdown, %{}}
     else
-      {:noreply, {id_map, table_name, user_count}}
+      state = %{
+        state
+        | connected_computers: connected_computers - 1
+      }
+
+      {:noreply, state}
     end
   end
 
-  @spec assemble_players(list(), map(), number()) :: list()
-  def assemble_players(seats, id_map, active_seat) do
-    seats
-    |> Enum.with_index(0)
-    |> Enum.map(fn {id, index} ->
-      %FakeArtist.Player{
-        id: id,
-        name: Map.get(id_map, id),
-        seat: index,
-        is_active: index == active_seat
-      }
-    end)
+  @spec update_player_role(map(), number(), atom()) :: map()
+  defp update_player_role(players, id, new_role) do
+    player = Map.get(players, id)
+    player_with_new_role = %{player | role: new_role}
+    Map.put(players, id, player_with_new_role)
+  end
+
+  @spec update_player_name(map(), number(), atom()) :: map()
+  defp update_player_name(players, id, new_name) do
+    player = Map.get(players, id)
+    player_with_new_name = %{player | name: new_name}
+    Map.put(players, id, player_with_new_name)
+  end
+
+  @spec get_active_players(map(), list()) :: list()
+  defp get_active_players(players, active_players) do
+    current_active_player_id = active_players |> hd()
```

Comment:
You don't have to change this, but I believe the community considers it "better style" to only use the pipeline operator when you have multiple fn calls that would otherwise be nested. Ie:
```ex
# good
hd(active_players)

# also good
active_players |> hd() |> String.upcase()

# bad
active_players |> hd()